src/crate/client/sqlalchemy/compiler.py

```diff
@@ -292,3 +292,8 @@ def limit_clause(self, select, **kw):
         Generate OFFSET / LIMIT clause, PostgreSQL-compatible.
         """
         return PGCompiler.limit_clause(self, select, **kw)
+
+    def for_update_clause(self, select, **kw):
+        # CrateDB does not support currently 'FOR UPDATE' statements.
+        # See https://github.com/crate/crate-python/issues/577
+        return ''
```

Comment:
I am always not sure if empty string or `None` should be returned here. It could actually be so so. See also GH-582 for comparison. Can you find out more about this detail?

Reply:
The thing is that `for_update_clause` is later used in https://github.com/sqlalchemy/sqlalchemy/blob/f96da1a8985dfb9303eef41135044c82420d08cc/lib/sqlalchemy/sql/compiler.py#L4919 in string concatenation, returning None would raise a `Type error`

This could be 'patched' if we override `SQLCompiler._compose_select_body` but then it might change this trivial change into changing and overriding some private parts of the class.

Reply:
Thanks for the elaboration. I think "empty string" is the right choice here, then. Mind adding a changelog item, so we can merge without further ado?

Reply:
Mind adding a changelog item, so we can merge without further ado?